src/helpers/binding.ts

```diff
@@ -0,0 +1,163 @@
+const placeholderRegExp =
+  /["'`](?:(?<=")[^"\\]*(?:\\.[^"\\]*)*"|(?<=')[^'\\]*(?:\\.[^'\\]*)*'|[^`]*`)|(\?\d*|[:@$][a-zA-Z_]\w+)/gmu;
```

Comment:
Yeh, so you'll note that this is using non-capturing groups for al the "escape" items (", ', `), and lookbehind to make sure that we're capturing everything between a matching set of either ", ', or `. The secondary quotes are all |'rd together _with_ the placeholder items (?, ?NN, :, @, %), so that we match _either_ an escaped quote, _or_ one of the placeholders. If we match on the string escapes, we short circuit the |s and we won't find them. This works so that we don't try to replace quoted ? and things like that. I'm sure there are edge cases that this will miss (for instance, I've now noticed I also have to add [] escaping), but give we'll eventually have real placeholder treatment and prepared statements, I think this works ok for now.